python/ray/serve/endpoint_state.py

```diff
@@ -33,3 +106,19 @@ def get_endpoints(self) -> Dict[EndpointTag, Dict[str, Any]]:
                 "shadows": shadow_dict,
             }
         return endpoints
+
+    def delete_endpoint(self, endpoint: EndpointTag) -> None:
+        # This method must be idempotent. We should validate that the
```

Comment:
If we choose to **not have internal locks**, we should at least pass in a lock and assert that it is held.